### PR TITLE
Add configurable staking URL to register command

### LINKED ISSUE #1741: Add Configurable URL Option for Staking Submission API
**Add Configurable URL Option for Staking Submission API**

---

**Description:**

Currently, the URL for submitting staking information to the staking API (`ssb.oxen.observer`) is hardcoded. To improve flexibility and adaptability in various environments, we should  make this URL configurable through a CLI option. This is marked as a TODO in the application.

---

**Task Breakdown:**

- **Refactor the submission URL:**  
  Replace the hardcoded `https://ssb.oxen.observer/api/store/{}` URL (`src/daemon/rpc_command_executor.cpp:2785`) with a configurable option from the application's configuration.
  
- **Add new config option:**  
  Introduce a new configuration entry (e.g., `staking_api_url`) that will allow the user to specify the base URL for the staking submission API.

- **Fallback mechanism:**  
  Ensure that the code falls back to the current hardcoded URL if the new config option is not set, maintaining backward compatibility.

- **Update the code:**  
  Modify the logic within the submission process to use the URL from the configuration, replacing:
  ```cpp
  auto url = cpr::Url{"https://ssb.oxen.observer/api/store/{}"_format(snode_pubkey)};
  ```
  with the configurable URL.

- **Testing:**
  - Ensure that the submission works correctly using both the default hardcoded URL and a user-specified URL.
  - Test various edge cases like invalid URLs or missing configuration.

---

**Acceptance Criteria:**

1. The API URL for staking submission is configurable via a config option.
2. The application falls back to the current hardcoded URL if the config option is not set.
3. The submission process behaves the same with both the default and custom URLs.
4. Error handling for incorrect or missing URLs is implemented.

### PR DESCRIPTION
**Pull request by jagerman**
Fixes #1741 

(NB: this is on top of #1753, but only the last two commits here are relevant).

- rename overly verbose `prepare_eth_registration` to just `register`

- `register` now takes an optional URL to submit to, usage is:
    - `register 0xabc... https://example.com`/
    - This submits to https://example.com/api/store/PUBKEY, and then
      prints a message to visit https://example.com/register/PUBKEY

- `register` now defaults to stake.getsession.org, rather than
  ssb.oxen.observer, and the stake.getsession.org is now unified for
  both backend and frontend.  (Previously it submitted to ssb, but
  printed misleadingly that it was submitting to stake.getsession.org).

- added colored formatting for the "register ... print" output, in
  particular using alternating regular/bright colours for each
  64-hex-digit segment to make it a lot easier to split it up into
  multiple u256 values as the contracts require.

- fixed the register command not giving an error if oxend is not a
  service node.  Previously it was outputting 0 pubkeys.

- fixed the BLS_REGISTRATION_REQUEST endpoint returning zero pubkeys
  when not a service node instead of producing an error.

- fixed `register` error output to be a bright red error message rather
  than a error log line.
  
Edit: added in last commit:
- makes the default staking URL a network-dependent option.

- renames `bls_registration_request` endpoint to `contract_registration` as the former name was misleading, along with tweaking the returned parameters to be more similar to related endpoints.

- adds mostly-internal-user `plaintext` and `contract` print modes for the `register` command (specified instead of `print`, but deliberately omitted from the public usage info): `plaintext` is like `print` but without formatting; `contract` splits up and formats values as needed by the contract, for easier direct contract interactions.

### CODE REVIEW

File: src/daemon/rpc_command_executor.cpp [hunk 1]
```diff
@@ -2768,42 +2762,58 @@ bool rpc_command_executor::prepare_eth_registration(std::string_view operator_ad
...
+                "{}\n",
+                fmt::styled("L2 Contract Registration Information:", fmt::emphasis::underline));
         tools::msg_writer(
-                "Submitting L2 staking information to staking.getsession.org, please wait.");
-        // TODO: make config option
-        auto url = cpr::Url{"https://ssb.oxen.observer/api/store/{}"_format(snode_pubkey)};
+                "{} {}\n"
+                "{} {}{}\n"
+                "{} {}{}\n"
+                "{} {}{}{}{}\n"
```
Comment (Doy-lee) [inline src/daemon/rpc_command_executor.cpp]: I know there's colours here but potentially adding a space between the 32 byte chunks is more useful because it means when you double click on the word-chunk the word-selection will be delimited by the space and select just the 32 byte chunk instead of the entire word.

This makes the piece-meal copy and paste into the 32 byte input text fields on the blockchain explorers easier than having to drag the cursor over the word and delimit the selection yourself based off the colour of the chunk.

**Review by Doy-lee — APPROVED** [review]
The two new commits look good, I have questions about the base async stuff in the other PR.

At a higher level there's was some meta discussions about changing the command name as `register` currently conflicts with the frontend's re-use of `register` to actually push the node onto the contract and SNL, but this is not a blocker as we can easily change it later down the line.

This means in future updating `register` and `contract_registration` endpoints for consistency.

File: src/daemon/rpc_command_executor.cpp (hunk 1 above)
Comment (jagerman) [reply src/daemon/rpc_command_executor.cpp]: Good idea; updated.
